src/ansys/tools/visualization_interface/backends/pyvista/pyvista.py

```diff
@@ -324,8 +315,10 @@ def hover_callback(self, _widget, event_name) -> None:
         renderer = plotter.iren.get_poked_renderer(x, y)
         self._hover_picker.Pick(x, y, 0, renderer)
         actor = self._hover_picker.GetActor()
-        if event_name == "TimerEvent" and actor is not None and actor in self._object_to_actors_map:
+        if actor is not None and actor in self._object_to_actors_map:
             custom_object = self._object_to_actors_map[actor]
+            for label in self._added_hover_labels:
+                self._pl.scene.remove_actor(label)
```

Comment:
Why remove the actor here?

Reply:
If you hover from one actor to another without passing through no actors for some seconds, labels from previous actors will still be showing.

Reply:
Oohh.. clever. Got it.